pkg/networkserver/downlink.go

```diff
@@ -193,7 +193,27 @@ outer:
 		down, dev.QueuedApplicationDownlinks = dev.QueuedApplicationDownlinks[0], dev.QueuedApplicationDownlinks[1:]
 
 		if len(down.FRMPayload) > int(maxDownLen) {
-			logger.Warn("Application downlink present, but the payload is too long, skipping...")
+			logger.Warn("Application downlink present, but the payload is too long, informing Application Server...")
+			ok, err := ns.handleASUplink(ctx, dev.EndDeviceIdentifiers.ApplicationIdentifiers, &ttnpb.ApplicationUp{
+				EndDeviceIdentifiers: dev.EndDeviceIdentifiers,
+				CorrelationIDs:       append(events.CorrelationIDsFromContext(ctx), down.CorrelationIDs...),
+				Up: &ttnpb.ApplicationUp_DownlinkFailed{
+					DownlinkFailed: &ttnpb.ApplicationDownlinkFailed{
+						ApplicationDownlink: *down,
+						Error: ttnpb.ErrorDetails{
```

Comment:
I think there's a util for doing this

Reply:
I also thought so, but I did not manage to find it

Reply:
Isn't it `errors.ErrorDetailsToProto` ?